frame/elections-phragmen/src/lib.rs

```diff
@@ -1153,10 +1170,89 @@ impl<T: Config> ContainsLengthBound for Pallet<T> {
 	}
 }
 
+
+impl<T: Config> ElectionDataProvider for Pallet<T> {
+	type AccountId = T::AccountId;
+	type BlockNumber = T::BlockNumber;
+	type MaxVotesPerVoter = ConstU32<MAXIMUM_VOTE>;
+
+	fn electable_targets(
+		maybe_max_len: Option<usize>,
+	) -> frame_election_provider_support::data_provider::Result<Vec<Self::AccountId>> {
+			let mut candidates_and_deposit = Self::candidates();
+			// add all the previous members and runners-up as candidates as well.
+			candidates_and_deposit.append(&mut Self::implicit_candidates_with_deposit());
```

Comment:
What is `implicit_candidates_with_deposit` here? 

I recall now that all candidates are considered to be voters, and now since they are two distinct functions, we would need to sync this somehow, or re-read them etc.

I am more and more leaning toward `type Solver: NPoSSolver` was a better approach :(